### ISSUE TITLE
Composite Product Sync Issues

### ISSUE THREAD

**Issue by jlconrad3** [facebookincubator/facebook-for-woocommerce#743]
I created a composite Woocommerce product, but it's not synced.  Here's a link to my shop: https://www.facebook.com/pg/TheCatholicWoodworker/shop

Let me know if I need to make any product changes.  Thanks!

**Comment by Luther2010** [comment]
## Automated Support Message

Hello,

Thanks for your interest in our plugin! Our developer team tries to provide personal support on GitHub whenever possible.

If you have a feature request or feedback for the developers, reply with `Feature Request` or `Feedback`.
If you are a developer or have a question for the developers related to code, reply with `Developer Question`.
If you are running ads with us, reply with `Advertiser`.

If you have encountered a bug or a problem with the plugin, keep reading for information that can help you resolve the issue yourself or provide us details we need to help you.

## Search Existing Issues

First, please [search our issues](https://github.com/facebookincubator/facebook-for-woocommerce/issues?q=) to see if your problem has been reported before.
There may already be a workaround or a solution to your problem. You can use the [search bar](https://github.com/facebookincubator/facebook-for-woocommerce/issues?q=) near the top of our issues page. Be sure to search both _open_ and _closed_ issues.
If you do find a similar issue, leave a comment on that issue instead of creating a new one. Your comment helps us prioritize which existing issues are important and urgent.

## Common Resolution Methods

If you are experiencing a problem or bug and have not found any similar issue, please try the following:

* Upgrade to the latest version of the plugin
* Try to `Resync` products using the button on the plugin config page
* Reset the plugin by selecting `Reconfigure > Advanced Options > Delete`, and setting up this plugin again
* Try deactivating other plugins that you think may be interfering, then reset and set up again

## Necessary Information

If the above steps did not help, please provide the following information in your next comment in order for us to debug your problem:

* WooCommerce version
* WordPress version
* Plugin version (please note: we only support up-to-date versions of the plugin)
* A screenshot of where you see the problem (for example: if it is on Facebook, provide a screenshot of Facebook, if it is on WooCommerce, provide a screenshot of WooCommerce. You can provide multiple screenshots if the problem relates to different screens.)
* The URL of your website
* The URL of your Facebook Page
* The URL(s) of where you took your screenshot(s)
* If applicable, detailed steps you took in order to see the problem

Please include `Debug Ready` in your reply so our automated system can note your issue for manual review. Please be aware that due to a high volume of issues, we may not be able to respond immediately.

## Response Time

We prioritize issues with the tags `Advertiser` or `Multiple Reports`.
For issues that have the tags `Debug Ready`, `Feature Request` or `Developer Question`, we aim to respond within 30 days.

If you have not responded to our automated message with one of the above phrases, such as `Debug Ready`, we may close your issue after 30 days. If a `Feature Request` is inactive (without new requests, messages or pull requests) for 6 months, we may close it.

Please note that we are not always able to provide support for issues related to interaction from other plugins with this plugin or related to the Facebook advertising interface. To get support related to Facebook ads, try visiting the [Facebook Ads Help Center](https://www.facebook.com/business/help).

For more information on our response times, visit [Facebook Support on GitHub](https://github.com/facebookincubator/facebook-for-woocommerce/wiki/Facebook-Support-on-GitHub).

Thank you,
The Facebook Dev Team

**Comment by diego-r-a** [comment]
I have the same problem. "Composite Products" (Woocommerce official plugin) does not sync. They are marked as synced but does not appear in the catalog. They are also assigned a "Facebook ID" according to the plugin, but again they don't show in the Catalog. "Simple Products" are correctly cataloged.

Wordpress: 4.9.8
Woocommerce: 3.5.1
Facebook for Woocommerce: 1.9.6
Woocommerce Composite Products: 3.14.2

**Comment by jameskim0903** [comment]
Hi @jlconrad3 @diego-r-a, I tried creating composite product using Woocommerce Composite Products v3.14.2.
I successfully synced the composite product to Facebook page (Facebook for Woocommerce: 1.9.7), and see the product in FB Page Shop section. There must be some other issue that is not related to "Composite Product". Do you have the product set to "Visible" where you see Facebook ID? Also, can you share the Facebook ID that is assigned to your composite product so that we can further drill down to the issue?

Thanks

**Comment by jlconrad3** [comment]
No, I can't click visible because it hasn't been added to Facebook yet.  It
just says, "This product is not yet synced to Facebook."  For all other
products, once I click update (after the plug-in is installed) they
automatically get added and I can click the "Visible" box.

**Comment by diego-r-a** [comment]
Hi @jameskim0903  , I made a few more tests. The product is synced and appears in the catalog ONLY if you assign a default price to the Composite Product. If you don't assign a default price, so that the Composite Product price is determined by it's "Components" configuration, the product does not appear in the Catalog.
As @jlconrad3 said, the product is not synced by default, but you can "force" sync it by doing a "quick edit" and updating.
But even if you assign a "default" price to the Composite Product and it's synced, it does not show the correct price in Facebook since the plugin only "reads" this "default" price and does not do the composite price "calculation" including the "components" like Woocommerce does, i guess that's why it does not show in the catalog if you don't fill the "default price", the product is probably being synced with no price.
Just to shed a final light on this, the correction is probably very simple since you would just need a 'if' to check is the product type is 'composite' and use the function get_composite_price() instead of get_price() or whatever you are using.
Thanks @jameskim0903 for the attention given to this issue. I hope i could help! Here is a Facebook ID for a Composite Product that does not show in the Catalog as you asked: 2768069983219145

**Comment by diego-r-a** [comment]
Hi @jameskim0903 , me again. 
I just did exactly what i said in the last comment. I added a few lines in fbproduct.php right after the 'If it's a booklable product' conditions and actions:
if (class_exists('WC_Product_Composite') && $this->is_type( 'composite' )) {
$regular_price = $this->get_composite_price();
}
and now the Composite products are finally showing in the Catalog with their real 'initial configuration' price just like in Woocommerce. I did it really fast and 'lazy' with this code because that's just what i needed, but you probably should do the full implementation with get_composite_regular_price() and other things to get the 'sale' price and stuff.  Hope you implement something in the future so i don't lose my code with an update! lol
I guess i should probably do a Pull request, right?
Thanks again.
PS.: The Facebook ID for the product that i gave you in the last comment in probably not usefull for you anymore since i edited the code and made the plugin send the product price as stated above.

**Comment by Git-Good** [comment]
Hi @jameskim0903, has the change been included in the codebase yet? I am still experiencing the problem described in this issue.

**Comment by sassoli-nier** [comment]
I have the same problem, I think this patch has not been merget yet...
